src/formatDate.js

```diff
@@ -8,7 +8,86 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let result = '';
+  const oldDate = date;
+  const newFormat = toFormat;
+  let array = [];
+  let year = '';
+  let yearBig = '';
+  const newArray = [];
+  let indexYear = 0;
+  let indexYearbig = 0;
+
+  const [, , , oldSeparator] = fromFormat;
+  const [, , , newSeparator] = newFormat;
+
+  array = oldDate.split(oldSeparator);
+
+  if (newFormat.includes('YYYY') && fromFormat.includes('YYYY')) {
+    for (const n of newFormat) {
+      const searchInd = fromFormat.indexOf(n);
+
+      newArray.push(array[searchInd]);
+    }
+    newArray.pop();
+    result = newArray.join(newSeparator);
+
+    return result;
+  }
```

Comment:
The logic here assumes that both 'YYYY' are present in `newFormat` and `fromFormat`, but it doesn't handle cases where they might not be aligned correctly. Consider checking the alignment of formats before proceeding.